File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    r=[]
    # 기준 열 해시 코드
    th={'code':0,'date':1,'maximum':2,'remain':3}
    for i,d in enumerate(data):
        if d[th[ext]]<val_ext:    
            r+=[d]
    r.sort(key=lambda x:x[th[sort_by]])
    return r

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> data, string ext, int val_ext, string sort_by) {
    vector<vector<int>> r;
    map<string,int> th={{"code",0},{"date",1},{"maximum",2},{"remain",3}};
    for(int i=0;i<data.size();i++) {
        if(data[i][th[ext]] < val_ext) {
            r.push_back(data[i]);
        }
    }
    sort(r.begin(),r.end(),[&th,&sort_by](vector<int> &v1, vector<int> &v2) {
        return v1[th[sort_by]]<v2[th[sort_by]];
    });
    return r;
}